src/base/dev/vga/vgaemu.c

```diff
@@ -1497,8 +1500,10 @@ void vgaemu_reset_mapping()
       error("VGA: map failed at page %x\n", page);
       return;
     }
-    vgaemu_update_prot_cache(page, prot);
   }
+  /* for simplicity cache R/W prot on UMBs too */
```

Comment:
I really think this is a micro opt that
makes the code worth. The UMB
check that this removes, can better
be inlined, no?

Reply:
Well, I think I have another approach here: in revert this change but instead in `vgaemu_prot_ok` don't use
```
  if(page >= 0xa0 && page < 0xc0) {
    return vga.mem.prot_map0[page - 0xa0] == prot ? 1 : 0;
  }
```
but replace that check by one that excludes the UMBs.